### DISCUSSION TITLE
#!zh 开发常见问题答疑 #!FAQ

### DISCUSSION THREAD

**Discussion by ly525** [ly525/luban-h5#110] [General]
# 本地开发常见问题
1.  `API 403 Forbidden`： [#!zh: 解决方案](https://ly525.github.io/luban-h5/zh/getting-started/quick-start.html#_2-%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9)

2.  预览页面空白：[解决方案-构建预览所需的渲染模块](https://ly525.github.io/luban-h5/zh/getting-started/quick-start.html#_2-%E6%9E%84%E5%BB%BA%E9%A2%84%E8%A7%88%E6%89%80%E9%9C%80%E7%9A%84%E6%B8%B2%E6%9F%93%E5%BC%95%E6%93%8E)

# 安装依赖常见问题

> 依赖安装遇到的问题，99% 都能Google 搜索到
> 因此不再单独解答，有问题提issue，或者在下面评论

## 问题1：安装依赖卡住

```bash
[5/5] Building fresh packages...
[1/7] ⡀ sqlite3
[-/7] ⡀ waiting...
[-/7] ⡀ waiting...
[-/7] ⡀ waiting...
error C:\website\luban-h5\back-end\h5-api\node_modules\sqlite3: Command failed
```

解决方案：
sqlite3相关的东西被墙在外面了，虽然已经改了淘宝镜像，还是翻墙下了sqlite3

其实只要配置下 sqlite3的二进制镜像就可以了，还是没看 mirror.sh 导致的
里面已经写的很清楚了

```
npm set node-sqlite3-binary-host-mirror "https://npm.taobao.org/mirrors"
npm set sqlite3-binary-site "https://npm.taobao.org/mirrors/sqlite3"
```

## 问题2: 后端无法启动 (Could Not Start API Server)
> #!zh 现象：./luban-h5 init && ./luban-h5 start 之后，localhost:1337 无法访问
> #!en behavior: ./luban-h5 init && ./luban-h5 start,  localhost:1337 is not accessable

> [2020-05-13T06:16:34.456Z] debug ⛔️ Server wasn't able to start properly.
[2020-05-13T06:16:34.458Z] error ReferenceError: URL is not defined

### 解决方案：

升级 Node.js 版本 >= 10
更多请参见：https://github.com/strapi/strapi/issues/2428

> 鲁班的后端用的 strapi.js，后端开发过程中如果遇到什么问题
1. 优先看鲁班的文档
2. 搜索 strapi 的issues

---

## 不欢迎如下行为
1. 伸手党
2. 一上来就贴张报错截图，啥都不说<node、yarn版本、用yarn 还是 npm 安装的依、是不是配置了镜像、有没有配置科学上网>

### 解决问题的主要思路（一定要看）！！！！！！！！！！！！！！
仔细阅读，贴出来的东西都是有用的

1. 设置 npm 或 yarn 镜像，自动配置镜像脚本参见：https://github.com/ly525/luban-h5/blob/dev/deploy/mirror.sh，使用方法 拷贝下来，加上可执行权限，跑一下就行了。
会自动把yarn和npm 配置成为淘宝的镜像源

2. 鲁班的项目要用 yarn 安装依赖，不要用 npm ！！！文档已经写的很清楚了

3. 安装时可以装的上的，就是有些包在海外aws 上（原因看：https://github.com/ly525/luban-h5/blob/dev/deploy/mirror.sh 这里），导致时间很长。（可能超时了。自己弄个梯子，能直联 aws 就行了，h5-api 安装依赖耗时有时候会很长（平均耗时需要498.06s））

4. 建议切换成联通的网络，联通的出口带宽多一些（相对移动和电信）

5. 提问的时候，列出来自己的 node、yarn 版本，是否配置镜像等信息，提问的基本礼仪：https://zhuanlan.zhihu.com/p/19779979

### 补充资料
1. 前端项目建议移除 node-sass 依赖，替换为 dart-sass；node-sass是古老的东西了，sass 官方都不用了
```bash
yarn add sass # 安装 dart-sass，dart-sass 现在是 sass 的主要实现。没有 binary 依赖
```

### 相关 issue
#92 
#101 

> 补充说明：
> 相关问题，如果没有解答，可以下方提问
> 提问前，请先阅读：[如何提问，并获得高回复率——提问的智慧（精缩版）](https://zhuanlan.zhihu.com/p/19779979)
> 也可以到 [鲁班H5-吐个槽论坛](https://support.qq.com/products/93432?) 交流

**Comment by dncaowei** [comment]
你好 鲁班项目 我们试着在demo页面做了一个网页 如果想把做好的页面 部署到自己到域名下面，请问怎么导出这些资源

**Reply by ly525** [reply]
你希望导出什么呢？一个 zip 包吗？

**Reply by dncaowei** [reply]
对

**Reply by dncaowei** [reply]
类似这个项目里面的下载功能 http://show.lzuntalented.cn/#/list 您可以看一下 这个h5编辑没有你做的好

**Reply by ly525** [reply]
可以实现，原理是类似的。将这个h5 的index.html、css、js资源创建zip包，然后提供下载。

**Reply by lijialong1313** [reply]
所以是现在还没有可以下载的独立预览版本吗？

**Comment by bioo** [comment]
请问后期是否能支持表单组件化呢？

**Reply by ly525** [reply]
你的意思是：动态表单组件是吗？有可以参考的例子吗

**Comment by dncaowei** [comment]
你好 鲁班 我们试着在demo页面加了一个按钮 想在点击按钮的时候跳转到一个外部页面http的地址 这个有没有办法实现

**Reply by ly525** [reply]
嗯，最近会给按钮、图片统一加上跳转能力
